examples/chain_replication.pyv

```diff
@@ -0,0 +1,89 @@
+# node ids and states
+sort node_id
+sort node_state
+
+# ordering on node states
+immutable relation le(node_state, node_state)
+axiom le(X,X) # reflexive
+axiom le(X,Y) & le(Y,X) -> X = Y # anti-symmetric
+axiom le(X,Y) & le(Y,Z) -> le(X,Z) # transitive
+axiom le(X,Y) | le(Y,X) # total
+
+# empty state
+immutable constant empty: node_state
+
+# the empty state is the least state
+axiom le(empty, X)
+
+# let's say that there are three nodes
+immutable constant head: node_id
+immutable constant middle: node_id
+immutable constant tail: node_id
+
+# the head, middle, and tail nodes are all different
+# (this is too strong; it should be OK for them to all be the same, too)
+axiom head != middle & head != tail & middle != tail
```

Comment:
is it correct to say that this axiom is necessary for now because the state of each node is modeled as a separate mutable constant below? so if two nodes were equal, the model wouldn't make sense because one node would have two different states?

Reply:
I tweaked some things, and it seems like the axiom I actually need is:

```
# if head = tail, then head/middle/tail are all the same
axiom head = tail -> head = middle && middle = tail
```

Otherwise I get situations where head and tail are the same node but middle is a different node.